### ISSUE TITLE
DoNotHideBaseClassMethods shouldn't apply to constructors

### ISSUE THREAD

**Issue by svick** [dotnet/roslyn-analyzers#1691]
#### Analyzer package

Microsoft.CodeQuality.Analyzers, version [2.6.0](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers/2.6.0)

#### Analyzer

DoNotHideBaseClassMethods

#### Repro steps

1. `dotnet new classlib`
2. `dotnet add package Microsoft.CodeQuality.Analyzers`
3. Change Class1.cs to:

    ```c#
    class Base
    {
        public Base(string input) { }
    }

    class Derived : Base
    {
        public Derived(object input) : base(null) { }
    }
    ```
4. `dotnet build`

#### Expected behavior

No CA1061, since constructors are not inherited. In other words, if I removed the `Derived` constructor, I still wouldn't be able to write `new Derived("some string")`.

#### Actual behavior

```
> dotnet build
Microsoft (R) Build Engine version 15.5.180.51428 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restoring packages for C:\code\tmp\hwapp\hwapp.csproj...
  Generating MSBuild file C:\code\tmp\hwapp\obj\hwapp.csproj.nuget.g.props.
  Restore completed in 751,09 ms for C:\code\tmp\hwapp\hwapp.csproj.
Class1.cs(8,12): warning CA1061: Change or remove 'Derived.Derived(object)' because it hides a more specific base class method: 'Base.Base(string)'. [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(6,7): warning CA1812: Derived is an internal class that is apparently never instantiated. If so, remove the code from the assembly. If this class is intended to contain only static members, make
it static (Shared in Visual Basic). [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(8,27): warning CA1801: Parameter input of method .ctor is never used. Remove the parameter or use it in the method body. [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(3,24): warning CA1801: Parameter input of method .ctor is never used. Remove the parameter or use it in the method body. [C:\code\tmp\hwapp\hwapp.csproj]
  hwapp -> C:\code\tmp\hwapp\bin\Debug\netstandard2.0\hwapp.dll

Build succeeded.

Class1.cs(8,12): warning CA1061: Change or remove 'Derived.Derived(object)' because it hides a more specific base class method: 'Base.Base(string)'. [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(6,7): warning CA1812: Derived is an internal class that is apparently never instantiated. If so, remove the code from the assembly. If this class is intended to contain only static members, make
it static (Shared in Visual Basic). [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(8,27): warning CA1801: Parameter input of method .ctor is never used. Remove the parameter or use it in the method body. [C:\code\tmp\hwapp\hwapp.csproj]
Class1.cs(3,24): warning CA1801: Parameter input of method .ctor is never used. Remove the parameter or use it in the method body. [C:\code\tmp\hwapp\hwapp.csproj]
    4 Warning(s)
    0 Error(s)

Time Elapsed 00:00:11.72
```

**Comment by dnperfors** [comment]
I was looking into fixing this, and it doesn't seem that hard to fix.
My idea would be to bail out the analysis as soon as it is a constructor, just like when it is an overload, like [here](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotHideBaseClassMethods.cs#L51).

However since there is a 'Discussion' tag on this issue, I wonder if it is worth to already provide a PR?

**Comment by mavasani** [comment]
@dnperfors I verified that binary FxCop in VS2015 does not report this and also the documentation mentions the diagnostic applies to only methods. Feel free to work on a fix. Thanks.

**Comment by dnperfors** [comment]
I will start on it when I’m back from holiday...
